content/blog/using-nunjucks-with-asyncapi.md

```diff
@@ -0,0 +1,198 @@
+- Chaining filters
+- Includes
+- Macros
+
+## Nunjucks basics by example
+
+All examples shown in this post can be explored in action in below CodeSandbox project.
+
+<iframe src="https://codesandbox.io/embed/learning-nunjucks-wis89?autoresize=1&codemirror=1&fontsize=14&theme=dark" style="width:100%; height:500px; border:0; border-radius: 4px; overflow:hidden;" title="learning-nunjucks" allow="geolocation; microphone; camera; midi; vr; accelerometer; gyroscope; payment; ambient-light-sensor; encrypted-media; usb" sandbox="allow-modals allow-forms allow-popups allow-scripts allow-same-origin"></iframe>
+
+In this learning project, I created a simple Express app that handles super short documentation generated from the AsyncAPI file. It is just a small sample of things that you can get from AsyncAPI using Nunjucks.
+
+I picked Nunjucks here for a reason. AsyncAPI community maintains [a tool for generating](https://github.com/asyncapi/generator/) different things from the specification document, and it is using Nunjucks as a templating engine. This basically means, use my CodeSandbox to experiment with Nunjucks, but if you plan to build some serious template for AsyncAPI, do it with the [generator](https://github.com/asyncapi/generator/) or just reuse existing templates.
+
+### Variables declaration
+
+You can declare inside the template a variable, that helps you in cases like loops. Their great use case is the same as in programming. If you have a value that you use more than once, assign it to a variable.
+
+I used it to keep the name of the API:
+```html
+{% set apiName = asyncapi.info().title() %}
+```
+
+Then I could use it multiple times, for example in these sentences:
+```html
+<!-- Sentence 1 -->
+The {{ apiName }} is licensed under {{ asyncapi.info().license().name() }}.
+
+<!-- Sentence 2 -->
+<p>Here you can find a list of channels to which you can publish and <strong>{{ apiName }}</strong> is subscribed to:</p>
+```
+
+### Built-in filters
+
+Unlike other engines, Nunjucks comes with many built-in helpers, called filters. There are around 40 different. You can for example easily make a value all uppercase:
+
+```html
+<!-- server.protocol() value comes as all lowercase -->
+using {{ server.protocol() | upper }} protocol
+```
+ 
+### Creating custom filters
+
+Built-in filters are awesome, but sometimes you need to create your own filters, and obviously it is possible. In my example I had to build a filter that will help me to modify the `server.url()` value. 
+
+In AsyncAPI document you can specify a server that the application uses to publish and consume messages from. In the URL you are allowed to use variables like this: `test.mosquitto.org:{port}`. Such variable can be described with different level of details. You can provide a default value, and even an enum of values. 
+
+In my example, instead of the URL like `test.mosquitto.org:{port}`, I wanted to get fixed URL with proper port number taken from the document:
```

Comment:
```suggestion
In my example, instead of a URL like `test.mosquitto.org:{port}`, I wanted to get a fixed URL with a proper port number taken from the document:
```